examples/reading_events.rs

```diff
@@ -0,0 +1,188 @@
+    Ok(())
+}
+
+pub async fn read_stream_backwards(client: &Client) -> Result<()> {
+    // region reading-backwards
+    let options = ReadStreamOptions::default().origin(Origin::End);
+    let result = client.read_stream("some-stream", &options, All).await?;
+
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion reading-backwards
+
+    Ok(())
+}
+
+pub async fn read_from_all_stream(client: &Client) -> Result<()> {
+    // region read-from-all-stream
+    let mut events = client.read_all(&Default::default(), All).await?;
+    // endregion read-from-all-stream
+
+    // region read-from-all-stream-iterate
+    while let Some(event) = events.try_next().await? {
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion read-from-all-stream-iterate
+
+    Ok(())
+}
+
+pub async fn read_all_overriding_user_credentials(client: &Client) -> Result<()> {
+    // region read-from-all-stream
+    let options = ReadAllOptions::default().authenticated(Credentials::new("admin", "changeit"));
+    let events = client.read_all(&options, All).await?;
+    // endregion read-from-all-stream
+
+    Ok(())
+}
+
+pub async fn ignore_system_events(client: &Client) -> Result<()> {
+    // region ignore-system-events
+    let mut events = client.read_all(&Default::default(), All).await?;
+
+    while let Some(event) = events.try_next().await? {
+        if event.get_original_event().event_type.starts_with("$") {
+            continue;
+        }
+
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion ignore-system-events
+
+    Ok(())
+}
+
+pub async fn read_from_all_stream_backwards(client: &Client) -> Result<()> {
+    // region read-from-all-stream-backwards
+    let options = ReadAllOptions::default().origin(Origin::End);
+
+    let mut events = client.read_all(&options, All).await?;
+    // endregion read-from-all-stream-backwards
+
+    // region read-from-all-stream-iterate
+    while let Some(event) = events.try_next().await? {
+        println!("Event> {:?}", event.get_original_event());
+    }
+    // endregion read-from-all-stream-iterate
+
+    Ok(())
+}
+
+pub async fn filtering_out_system_events(client: &Client) -> Result<()> {
+    let mut events = client.read_all(&Default::default(), All).await?;
+
+    while let Some(event) = events.try_next().await? {
+        if !event.get_original_event().event_type.starts_with("$") {
+            continue;
+        }
+
+        println!("Event> {:?}", event.get_original_event());
+    }
+    Ok(())
+}
+
+pub async fn read_from_stream_resolving_link_tos(client: &Client) -> Result<()> {
+    // region read-from-all-stream-resolving-link-Tos
+    let options = ReadAllOptions::default().resolve_links();
```

Comment:
**SUGGESTION:** Can we rename this to `resolve_link_tos` as in other clients?

Reply:
I found `resolve_link_tos` confusing and would advise renaming other clients. However, I'm ok for renaming it `resolve_link_tos`

Reply:
I think that this name is not ideal, but I'd prefer to keep consistency. We agreed with @George-Payne to follow the current naming convention from the .NET client, whereas it's possible. In the future, we should be careful while designing features to make sure that we use the right and intuitive naming.

Reply:
done